substrate/frame/babe/src/lib.rs

```diff
@@ -935,11 +935,14 @@ impl<T: Config> OnTimestampSet<T::Moment> for Pallet<T> {
 		let timestamp_slot = moment / slot_duration;
 		let timestamp_slot = Slot::from(timestamp_slot.saturated_into::<u64>());
 
-		assert_eq!(
-			CurrentSlot::<T>::get(),
-			timestamp_slot,
-			"Timestamp slot must match `CurrentSlot`"
-		);
+		if CurrentSlot::<T>::get() != timestamp_slot {
+			log::warn!(
+				target: LOG_TARGET,
+				"Timestamp slot {:?} does not match `CurrentSlot` {:?}!",
+				timestamp_slot,
+				CurrentSlot::<T>::get(),
+			);
+		}
```

Comment:
This needs to be reverted. What you want to do is to write some migration in your runtime to set these variables, but just blindly removing this check is not the solution

Reply:
I tried initializing this variable in my migration based on the Aura pallet `CurrentSlot`, and also timestamp pallet's `CurrentSlot`. 

The result is flakey, with the assert_ok! sometimes being triggered, depending on the block time. I didn't investigate too much, because the correct value for `CurrentSlot` is set after the first Babe block is mined anyway. 

I'm not 100% sure why we can't initialize based on Aura or Timestamp, but suspect it's because the value to set depends on the time between the Aura node spinning down and Babe node spinning up.

I would like to push back on it needing to be reverted anyways.. Why would it ever be helpful to panic in the runtime instead of log a warning, especially for what might be a non-critical issue?